internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -78,27 +88,95 @@ func (p *processRunner) Close() error {
 }
 
 type commandWrapper struct {
-	cmd              *exec.Cmd
+	launcher         *cap.Launcher
+	ctty             optional.Optional[int]
+	cgroupFd         int
+	stdin            uintptr
+	stdout           uintptr
+	stderr           uintptr
 	afterStart       func()
 	afterTermination func() error
 }
 
-//nolint:gocyclo
-func (p *processRunner) build() (commandWrapper, error) {
-	args := []string{
-		fmt.Sprintf("-name=%s", p.args.ID),
-		fmt.Sprintf("-dropped-caps=%s", strings.Join(p.opts.DroppedCapabilities, ",")),
-		fmt.Sprintf("-cgroup-path=%s", cgroup.Path(p.opts.CgroupPath)),
-		fmt.Sprintf("-oom-score=%d", p.opts.OOMScoreAdj),
-		fmt.Sprintf("-uid=%d", p.opts.UID),
+func dropCaps(droppedCapabilities []string, launcher *cap.Launcher) error {
+	droppedCaps := strings.Join(droppedCapabilities, ",")
+
+	if droppedCaps != "" {
+		caps := strings.Split(droppedCaps, ",")
+		dropCaps := xslices.Map(caps, func(c string) cap.Value {
+			capability, capErr := cap.FromName(c)
+			if capErr != nil {
+				fmt.Printf("failed to parse capability: %s", capErr)
```

Comment:
my understanding is that should never happen, as `cap.FromName` is statically defined, so we could even `panic` here